FaissMask/Internal/IndexSafeHandle.cs

```diff
@@ -79,17 +81,19 @@ protected override bool ReleaseHandle()
             return true;
         }
 
-        public float[] ReconstructVector(long key, int dimensions)
+        public float[] ReconstructVector(long key)
         {
+            var dimensions = NativeMethods.faiss_Index_d(this);
             var vector = new float[dimensions];
             NativeMethods.faiss_Index_reconstruct(this, key, vector);
             return vector;
         }
 
-        public float[][] ReconstructVectors(long startKey, long amount, int dimensions)
+        public float[][] ReconstructVectors(long startKey, long amount)
         {
             // TODO: There's probably a better way to marshall this 2D-array
             // Create one big float[] of the necessary size
+            var dimensions = NativeMethods.faiss_Index_d(this);
```

Comment:
```suggestion
            var dimensions = Dimensions;
```
Can we re-use the property definition?